File: fix_ids.py
import sqlite3

conn = sqlite3.connect('veritabani/site_yonetim.sqlite')
cursor = conn.cursor()

cursor.execute("SELECT id, blok, daire_no FROM daireler")
daireler = cursor.fetchall()

def sort_key(d):
    # admin account should probably be first (id=1)
    if not d[1]: 
        return ("", 0)
        
    blok = str(d[1])
    try:
        daire_no = int(d[2])
    except:
        daire_no = 0
    return (blok, daire_no)

daireler.sort(key=sort_key)

id_mapping = {}
for i, d in enumerate(daireler):
    id_mapping[d[0]] = i + 1

# Step 1: Assign temporary IDs
for old_id, new_id in id_mapping.items():
    temp_id = old_id + 100000
    if temp_id < 0: temp_id = -temp_id + 200000 # Handle already negative ones
    cursor.execute("UPDATE daireler SET id = ? WHERE id = ?", (temp_id, old_id))
    cursor.execute("UPDATE aidatlar SET daire_id = ? WHERE daire_id = ?", (temp_id, old_id))

# Step 2: Assign final IDs
for old_id, new_id in id_mapping.items():
    temp_id = old_id + 100000
    if temp_id < 0: temp_id = -temp_id + 200000
    cursor.execute("UPDATE daireler SET id = ? WHERE id = ?", (new_id, temp_id))
    cursor.execute("UPDATE aidatlar SET daire_id = ? WHERE daire_id = ?", (new_id, temp_id))

conn.commit()
print("IDs sorted successfully!")
conn.close()

File: fix_ids.js
const sqlite3 = require('sqlite3').verbose();
const db = new sqlite3.Database('veritabani/site_yonetim.sqlite');

db.serialize(() => {
    db.run("BEGIN TRANSACTION");

    db.all("SELECT * FROM daireler", [], (err, daireler) => {
        if (err) throw err;

        // Sort the daireler
        daireler.sort((a, b) => {
            if (a.blok !== b.blok) return a.blok.localeCompare(b.blok);
            return parseInt(a.daire_no) - parseInt(b.daire_no);
        });

        // Create a mapping from old id to new id
        const idMapping = {};
        let newId = 1;
        
        for (const daire of daireler) {
            idMapping[daire.id] = newId++;
        }

        // 1. Move daireler to temporary negative IDs to avoid unique constraint violations
        for (const daire of daireler) {
            const oldId = daire.id;
            const tempId = -oldId;
            db.run("UPDATE daireler SET id = ? WHERE id = ?", [tempId, oldId]);
            db.run("UPDATE aidatlar SET daire_id = ? WHERE daire_id = ?", [tempId, oldId]);
        }

        // 2. Move from temporary IDs to final IDs
        for (const daire of daireler) {
            const oldId = daire.id;
            const tempId = -oldId;
            const finalId = idMapping[oldId];
            
            db.run("UPDATE daireler SET id = ? WHERE id = ?", [finalId, tempId]);
            db.run("UPDATE aidatlar SET daire_id = ? WHERE daire_id = ?", [finalId, tempId]);
        }

        db.run("COMMIT", (err) => {
            if (err) {
                console.error("COMMIT FAILED:", err);
            } else {
                console.log("IDs successfully sorted and updated!");
            }
            db.close();
        });
    });
});
